app/src/main/java/org/oppia/app/profileprogress/RecentlyPlayedStorySummaryViewModel.kt

```diff
@@ -1,9 +1,29 @@
 package org.oppia.app.profileprogress
 
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.app.home.RouteToTopicPlayStoryListener
 import org.oppia.app.model.PromotedStory
+import org.oppia.app.shim.IntentFactoryShim
 
 /** Recently played item [ViewModel] for the recycler view in [ProfileProgressFragment]. */
 class RecentlyPlayedStorySummaryViewModel(
+  val activity: AppCompatActivity,
+  val internalProfileId: Int,
   val promotedStory: PromotedStory,
-  val entityType: String
-) : ProfileProgressItemViewModel()
+  val entityType: String,
+  private val IntentFactoryShim: IntentFactoryShim
+) : ProfileProgressItemViewModel(), RouteToTopicPlayStoryListener {
+  fun onStoryItemClicked() {
+    routeToTopicPlayStory(internalProfileId, promotedStory.topicId, promotedStory.storyId)
```

Comment:
Can't we just remove the interface `RouteToTopicPlayStoryListener` and its overridden function and copy its implementation here?  
```
  fun onStoryItemClicked() {
    val intent = IntentFactoryShim.createTopicPlayStoryActivityIntent(
      activity.applicationContext,
      internalProfileId,
      topicId,
      storyId
    )
    activity.startActivity(intent)
  }
```

Reply:
No actually this is an interm solution later on all these routes will be used once `IntentFactoryShim` is removed. So its better to keep it here so that we can easily track and update code wherever needed.